library/aes.c

```diff
@@ -1005,7 +1005,7 @@ int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
 #endif
 
 #if defined(MBEDTLS_PADLOCK_C) && defined(MBEDTLS_HAVE_X86)
-    if( aes_padlock_ace )
+    if( aes_padlock_ace > 0)
```

Comment:
`aes_padlock_ace` should always be initialized at this point since a successful call to `mbedtls_aes_setkey_enc` or `mbedtls_aes_setkey_dec` initializes it. Could you please clarify how you ended up with `aes_padlock_ace` set to -1 at this point? What build configuration are you using? If `MBEDTLS_PADLOCK_C` is enabled, do you have the corresponding hardware?

Reply:
Yes, the user should call mbedtls_aes_setkey_enc and mbedtls_aes_setkey_dec first with the right usage, but sometime the user may forget to call them and then the code normally crash inside the assembly code. Since most embedded developer isn't familiar with x86 assemble code, it's better to let's the code stop/crash inside the c source.

Reply:
Also, please put a short description in the commit message in the first line, with a more complete description in a subsequent paragraph. You wrote a very long first line which gets truncated in many interfaces. We generally like to follow [How to write a git commit message](https://chris.beams.io/posts/git-commit/), although we accept a subject line of up to 72 characters.

Reply:
Done.